src/test/java/net/imglib2/util/IntervalsTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * #%L
+ * ImgLib2: a general-purpose, multidimensional image processing library.
+ * %%
+ * Copyright (C) 2009 - 2018 Tobias Pietzsch, Stephan Preibisch, Stephan Saalfeld,
+ * John Bogovic, Albert Cardona, Barry DeZonia, Christian Dietz, Jan Funke,
+ * Aivar Grislis, Jonathan Hale, Grant Harris, Stefan Helfrich, Mark Hiner,
+ * Martin Horn, Steffen Jaensch, Lee Kamentsky, Larry Lindsey, Melissa Linkert,
+ * Mark Longair, Brian Northan, Nick Perry, Curtis Rueden, Johannes Schindelin,
+ * Jean-Yves Tinevez and Michael Zinsmaier.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imglib2.util;
+
+import org.junit.Test;
+
+import net.imglib2.Interval;
+import net.imglib2.test.ImgLib2Assert;
+
+public class IntervalsTest 
+{
+    @Test
+    public void testTranslateDim() {
+        final Interval input = Intervals.createMinMax( 1, 2, 3, 5, 7, 9 );
+        final Interval result = Intervals.translate( input, 10, 0 );
+        final Interval expected = Intervals.createMinMax( 11, 2, 3, 15, 7, 9 );
+        ImgLib2Assert.assertIntervalEquals( expected, result );
+    }
+
+    @Test
+    public void testTranslateUsesCorrectOverload() {
+        final Interval input = Intervals.createMinMax( 1, 2, 5, 7 );
+        // when the number of axes equals 2, then both translate() methods are applicable. 
+        // Which one is used?
+        final Interval result = Intervals.translate( input, 10, (long)1 );
```

Comment:
I've got one more question for you: `Intervals` has a `translate(interval, dimension, shiftAtDimension)` method already, and I added `translate(Interval interval, long... translation)`. When the interval is 2D, this means the number of arguments to both methods is the same, the only difference being `int` vs `long` for the 2nd argument.

I used `translate(Interval interval, long... translation)` because that matches the API of `Views.translate(RandomAccessibleInterval input, long... translation)`. But now that I noticed that signature clash I am wondering whether we should adjust the naming of one of the two methods... I'd prefer to rename the translation method that only translates a single dimension to something like `translateDim`, but that would break the current `Intervals` API.

Any opinions @maarzt @tpietzsch ?

Reply:
I just checked: inside imglib2 the existing `Intervals.translate(interval, dim, shift)` method is only used inside the `LoopPerformanceBenchmark`, but that doesn't mean nobody else is using it...

Reply:
I used the ```Intervals.translate(interval, dim, shift)``` in imglib2-algorithms. So it's very likely used in other places as well. It's terribly sad, we **must not rename** it. Because then the other translate method would chip in. And the method's behavior will be slightly different. It will introduce bugs, which are impossible to find and hard to notice.

The solution is to: change the signature ```Views.translate(RandomAccessibleInterval input, long... translation)``` to ```Views.translate(RandomAccessibleInterval input, long[] translation)```.

@chaubold In this PR please use ```Intervals.translate(Interval input, long[] translation)```. I will make a separate PR to change ```Views.translate```.

Reply:
Sorry I was wrong with the previous message there is no ```Views.translate(image, shift, dim)```

Reply:
@chaubold We don't need to fix this problem in this PR. Just create an issue.

Reply:
I created the issue